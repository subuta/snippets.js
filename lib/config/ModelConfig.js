import check from 'lib/_utils/apiCheck'

const ModelConfigShape = check.shape({
  schema: check.object
})

// eg:
// config = {
//   book: {
//     schema: {}
//   }
// }
export const validateModels = function (config) {
  return check.throw(check.objectOf(ModelConfigShape), config)
}

// eg:
// config = {
//   schema: {}
// }
export const validateModel = function (config) {
  return check.throw(ModelConfigShape, config)
}
