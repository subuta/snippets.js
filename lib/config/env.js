import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

export default (imports = []) => {
  const vars = _.zipObject(imports, imports)

  return build`
    // load .env first.
    require('dotenv').config()
    
    const ${s.raw(vars)} = process.env
    
    export ${s.raw(vars)}
    
    export default ${s.raw(vars)}
  `
}
