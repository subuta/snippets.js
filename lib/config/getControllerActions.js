import _ from 'lodash'

export default (config = {}) => {
  const {
    only, // the routes will generated.
    except  // the routes will ignored.
  } = config

  const routes = [
    'index',
    'show',
    'create',
    'update',
    'destroy'
  ]

  if (only) {
    return _.intersection(routes, only)
  } else if (except) {
    return _.difference(routes, except)
  }

  return routes
}
