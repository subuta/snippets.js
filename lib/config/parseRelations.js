import _ from 'lodash'
import pluralize from 'pluralize'

export default (config = {}) => {
  const { schema } = config

  const relations = _.transform(schema.relations, (result, value, key) => {
    let relatedModel = null;
    if (value.belongsTo) {
      relatedModel = pluralize.singular(_.toLower(value.belongsTo))
    } else if (value.hasOne) {
      relatedModel = pluralize.singular(_.toLower(value.hasOne))
    } else if (value.hasMany) {
      relatedModel = pluralize(_.toLower(value.hasMany))
    } else if (value.hasAndBelongsToMany) {
      relatedModel = pluralize(_.toLower(value.hasAndBelongsToMany))
    }
    result[key] = relatedModel
  }, {})

  return relations
}
