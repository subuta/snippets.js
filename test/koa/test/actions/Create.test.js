/* global expect, describe, it, beforeEach, afterEach */

import { build, format, snippets as s } from 'bld.js'

import CreateTest from 'lib/koa/test/actions/Create'

import { Routes as RoutesConfig, Models as ModelsConfig } from 'test/fixtures/config'

describe('koa/test/actions/Create', () => {
  beforeEach(() => {
  })

  afterEach(() => {
  })

  it('should create koa Create action test', () => {
    expect(format(CreateTest({
      model: 'user',
      routeConfig: RoutesConfig.user,
      modelConfig: ModelsConfig.user
    }))).toMatchSnapshot()
  })
})
