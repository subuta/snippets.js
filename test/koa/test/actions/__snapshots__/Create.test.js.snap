// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa/test/actions/Create should create koa Create action test 1`] = `
"test('post should create user', async (t) => {
  const {request, User} = t.context

  // mock jwks
  const token = createToken(privateKey, '123', currentUser)
  jwksEndpoint('http://localhost', [{pub: publicKey, kid: '123'}])

  const response = await request
    .post('/api/users')
    .set('Authorization', \`Bearer \${token}\`)
    .send({
      user: {
        id: 38169,
        auth0Id: 'card copying',
        locale: 'es',
        nickname: 'Georgette84',
        status: 'card copying',
        avatar:
          'https://s3.amazonaws.com/uifaces/faces/twitter/reabo101/128.jpg'
      }
    })

  t.is(response.status, 200)

  t.deepEqual(response.body.id, 38169)
  t.deepEqual(response.body.auth0Id, 'card copying')
  t.deepEqual(response.body.locale, 'es')
  t.deepEqual(response.body.nickname, 'Georgette84')
  t.deepEqual(response.body.status, 'card copying')
  t.deepEqual(
    response.body.avatar,
    'https://s3.amazonaws.com/uifaces/faces/twitter/reabo101/128.jpg'
  )
})
"
`;
